app/services/distributor_shipping_methods.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class DistributorShippingMethods
+  def self.shipping_methods(distributor:, customer: nil, checkout: false, apply_tags: true)
+    return [] if distributor.blank?
+
+    shipping_methods = distributor.shipping_methods
+    shipping_methods = shipping_methods.display_on_checkout if checkout
+    shipping_methods = shipping_methods.to_a
+
+    if apply_tags
+      OpenFoodNetwork::TagRuleApplicator.new(
+        distributor, "FilterShippingMethods", customer&.tag_list
+      ).filter!(shipping_methods)
+    end
+
+    shipping_methods.uniq
```

Comment:
At this point the `shipping_methods` is already an `Array` because the `ActiveRecord::Relation` fired a DB query on line 9 due to `to_a`. I'd be much better if we asked to remove duplicates to PostgreSQL which is much faster than asking Ruby to do it (and instantiate a bunch of bloated ActiveModel objects that are meant to be discarded). We could do that by using `#distinct`.

That, unfortunately, will only work when `apply_tags == false` when `true` `shipping_methods` will be an array at this point (making TagRuleApplication rely on AR relation is totally different topic) so we'll need to nest it within an `else` branch. I think it's worth it.